src/app/shared/view-mode.ts

```diff
@@ -0,0 +1,11 @@
+export enum SetViewMode {
+  List = 'list',
+  Grid = 'grid'
+}
+
+export enum ElementViewMode {
+  Full,
+  SetElement
+}
+
+export type ViewMode = SetViewMode | ElementViewMode;
```

Comment:
Please add comments to this file. I'm not sure I fully understand what this change is doing here, and why we needed to change `ViewMode` to `SetViewMode` everywhere.

Reply:
The viewmode we had before this PR determined how an entire set of DSOs should be rendered: e.g. render these search results as a list or as a grid.

This PR uses the same idea to determine which component to use for which entity-type in which situation. But at the level of the element, not the set. That way you can say: [this component](https://github.com/DSpace/dspace-angular/blob/ec6124035d35a505b6d6dec17305baa8220d3fd4/src/app/shared/object-list/item-list-element/entity-types/journal/journal-list-element.component.ts#L6) renders Journals when they're one element of a set, and [this component](https://github.com/DSpace/dspace-angular/blob/ec6124035d35a505b6d6dec17305baa8220d3fd4/src/app/+item-page/simple/entity-types/journal/journal-page-fields.component.ts#L14)  renders their full view. 

That's why we renamed the existing viewmode to SetViewMode, and the new one to ElementView Mode

Reply:
@artlowel : Ok, that makes more sense.  I'm fine with keeping the terminology here as "Set" vs "Element" as long as we add comments to this file to describe the various modes and terms and what they mean/represent.  I completely misunderstood this file when I looked at it initially :)